src/concurrency/timestamp_ordering_transaction_manager.cpp

```diff
@@ -624,6 +623,14 @@ ResultType TimestampOrderingTransactionManager::CommitTransaction(
   //// handle other isolation levels
   //////////////////////////////////////////////////////////
 
+  auto &rw_set = current_txn->GetReadWriteSet();
+
+  // if no modifying queries, treat as read-only
+  if (rw_set.empty()) {
+    EndTransaction(current_txn);
```

Comment:
I would just add a `LOG_TRACE` record here to record what is happening.

Reply:
Added